nums/core/array/blockarray.py

```diff
@@ -363,7 +366,13 @@ def _advanced_single_array_select(self, ss: tuple, axis: int = 0):
 
         src_arr = self
         np_ss = ss
-        ss = self.cm.put(ss)
+        ss = self.cm.put(
+            ss,
+            syskwargs={
+                "grid_entry": 0,
```

Comment:
@elibol Can you check this. I am not sure if the syskwargs are correct here

Reply:
Looks good, just set `"grid_shape": 1` so there's no confusion about what's going on. `ss` will be pulled to other nodes as needed.